### ISSUE TITLE
WNFileSystems has very rare hang on Linux.

### ISSUE THREAD

**Issue by AWoloszyn** [WNProject/WNFramework#85]
I saw a hang on Linux with WNFileSystems.
It was a release build so I do not have a ton of information, but I can say at least:

It was in get_scratch_path().
There was a directory in /tmp called scratch. (note: Nothing was on the end of this).
It may have been a weird case where rand() % returned 0 or a negative number?

I wonder if switching to something like 
```c++
      char name_template[9] = {0};
      for (size_t i = 0; i < 8; ++i) {
          name_template[i] = (rand()%10 + '0');
      }
```

would make more sense. (It may be more reliable? and we would always end up with 8-digit directories)

**Comment by devpow112** [comment]
Eventually I want to switch to something like cityhash to give me a random hash but for now I can look at switching to this. I may just switch this to be the behaviour on all platforms for consistency for now,

**Comment by devpow112** [comment]
Fixed in 4a621e33fa03f9ec62e73aed4e596d1513a64556
